compiler/modules/types.sexpr/schema.bal

```diff
@@ -0,0 +1,22 @@
+import wso2/nballerina.comm.sexpr;
+
+public type Field [sexpr:Str, Type];
+public type Atom ["list", Type[]] | ["list", Type[], int] | ["array", Type] | ["mapping", Type] | ["mapping", Field[], Type]; 
+// JBUG compile time stack overflow if ["!", Type] is added
+public type Type AtomRef | ["|", Type...] | ["&", Type...] | ["!", Type...] | Int | IntSubtype ; 
```

Comment:
Any reason for not using the already defined types?
```suggestion
public type Atom NonRepeatingTuple | RepeatingTuple | Array | Mapping | OpenMapping; 
// JBUG compile time stack overflow if ["!", Type] is added
public type Type AtomRef | Union | Intersection | Not | Int | IntSubtype ; 
```

Reply:
Ya it gives `invalid cyclic type reference in ...` error. I'll add a comment.
(@SasinduDilshara  please note)